okhttp/src/main/java/com/squareup/okhttp/internal/http/RouteSelectorImpl.java

```diff
@@ -45,7 +47,7 @@
  * choice of proxy server, IP address, and TLS mode. Connections may also be
  * recycled.
  */
-public final class RouteSelector {
+public final class RouteSelectorImpl implements RouteSelector {
```

Comment:
nit: rename to `RealRouteSelector` ? I really loathe the `Impl` suffix. (And want to nuke it from everything in OkHttp.)